### PR TITLE
[Bug]Alert plug-in: the get request in HTTP does not have a port

### LINKED ISSUE #11745: [Bug] [alert] The get method in the alarm plug-in HTTP does not have a port
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When sending a get request using the HTTP plug-in, it cannot be sent to the specified port

### What you expected to happen

When the code splices the URL, it does not bring the configured port, resulting in all requests reaching port 80

### How to reproduce

Simulate the alarm request in DS, configure the get request, and finally view the alarm log in logs under DS file. It is found that no matter how configured, there are only 80 ports

### Anything else

_No response_

### Version

3.0.0

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Jeffrey12333**
**Purpose of the pull request**

This pr will close #11745

### CODE REVIEW

**Comment by caishunfeng** [comment]
Hi @Jeffrey12333 can you add some UT case for it?

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java [hunk 1]
```diff
@@ -112,7 +112,8 @@ private void createHttpRequest(String msg) throws MalformedURLException, URISynt
             //GET request add param in url
             setMsgInUrl(msg);
             URL unencodeUrl = new URL(url);
-            URI uri = new URI(unencodeUrl.getProtocol(), unencodeUrl.getHost(), unencodeUrl.getPath(), unencodeUrl.getQuery(), null);
+            URI uri = new URI(unencodeUrl.getProtocol(), unencodeUrl.getPort() == -1 ? unencodeUrl.getHost()
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java:115]: Can we directly use `new URL(url).toURI()` ?

Comment (Jeffrey12333) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java:115]: yes， of cource

**Comment by Jeffrey12333** [comment]
> Hi @Jeffrey12333 can you add some UT case for it?

> 

> Hi @Jeffrey12333 can you add some UT case for it?

Sorry, this is my first submission. I don't know how to do it

**Comment by caishunfeng** [comment]
> > Hi @Jeffrey12333 can you add some UT case for it?
> 
> Sorry, this is my first submission. I don't know how to do it

see https://dolphinscheduler.apache.org/zh-cn/docs/dev/user_doc/contribute/join/unit-test.html

**Comment by Jeffrey12333** [comment]
> > > Hi @Jeffrey12333 can you add some UT case for it?
> > 
> > 
> > Sorry, this is my first submission. I don't know how to do it
> 
> see https://dolphinscheduler.apache.org/zh-cn/docs/dev/user_doc/contribute/join/unit-test.html

Yes, thank you

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java:115]: It's better to use `URIBuilder` to create the Http URI, don't connect the params by setMsgInUrl.

**Comment by EricGao888** [comment]
Could u plz rebase the code and resolve the conflicts? Thanks : ) @Jeffrey12333

**Pull request closed without merging**